Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
So there are three common cases, I think:

* Bindings like this: we can use whatever python so system python is fine; it's a bug if it matters.
* Applications (mostly in virtualenvs): I remember now (sorry) that we did discuss which python we ought to use for these. The most interesting argument for avoiding system Python here is that anything that uses the stdlib ssl module ends up using the system OpenSSL 0.9.8, which, chief among its faults, doesn't support TLS 1.2. I recall that I proposed using Homebrew's python for these, which was approved but didn't get fully rolled out. I later proposed shipping a separate keg-only homebrew-python-runtime formula that these could use to avoid confusing users who didn't expect the python in $PATH to change. That proposal expired.
* Embedders like vim: I don't think there are any reasonable options because people will want to be able to `import` the packages they've installed with their default pip and we can't accommodate that with bottles; the status quo (usually install Homebrew's python) is bad but I don't think other options are necessarily better. I think we used to default to system Python and that seemed also probably fine but it seems more stable since we switched.